maze.pl

```diff
@@ -136,9 +163,32 @@
 % Ensure that the tests pass with no warnings.
 %
 
+% (TODO): find what is wrong with path to self.
+%path_to(From,To,Path) :- connected(From,To); connected(To,From);
+%                         connected(To,Path);
+%			 connected(From,Path).
+
+
+%path_to(X,Y,[X,Y]):-corridor(X,Y),!.
+%path_to(X,Y,[X|R]):-corridor(X,Z),path_to(Z,Y,R),!.
+%
+
+path_to(From,To,Path) :-
+	travel(From,To,[From],Q),
+	rev(Q,Path),!.
 
+travel(From,To,Path,[To|Path]) :-
+	connected(From,To),!.
 
-:- begin_tests(path_to, [blocked('part 2, step 2')]).
+
+travel(From,To,Visited,Path) :-
+	connected(From,C),
+	C\==To,
+	not(mem(C,Visited)),
+	travel(C,To,[C|Visited],Path),!.
+
+
+:- begin_tests(path_to).
 
 % verify that a path from a cell to itself consists of just the one cell
 test(path_to_self) :-
```

Comment:
The reason why this test was unable to pass is because you start your accumulator with a non-empty path that you try to append to.